packages/common/core-utils/src/assert.ts

```diff
@@ -5,13 +5,25 @@
 
 /**
  * A browser friendly assert library.
- * Use this instead of the 'assert' package, which has a big impact on bundle sizes.
+ *
  * @param condition - The condition that should be true, if the condition is false an error will be thrown.
  * Only use this API when `false` indicates a logic error in the problem and thus a bug that should be fixed.
  * @param message - The message to include in the error when the condition does not hold.
  * A number should not be specified manually: use a string.
  * Before a release, policy-check should be run, which will convert any asserts still using strings to
  * use numbered error codes instead.
+ * @remarks
+ * Use this instead of the node 'assert' package, which requires polyfills and has a big impact on bundle sizes.
+ *
+ * Asserts using this API will be included in all configurations: there is no option to disable or optimize them out.
+ * Thus this API is suitable for detecting conditions that should terminate the application and produce a useful diagnostic message.
+ * It can be used to ensure bad states are detected early and avoid data corruption or harder to debug errors.
+ *
+ * In cases where the assert is very unlikely to have an impact on production code but is still useful as documentation and for debugging, consider using `debugAssert` instead
```

Comment:
Is this one of those scenarios where we can't use `{@link debugAssert}` because it's internal and this one is not?